apps/checkout-app/pages/api/pay.ts

```diff
@@ -0,0 +1,80 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { createMolliePayment } from "@/backend/payments/providers/mollie";
+import { createOrder } from "@/backend/payments/createOrder";
+import { allowCors } from "@/backend/utils";
+
+export type PaymentProviders = "mollie";
+const paymentProviders: PaymentProviders[] = ["mollie"];
+
+export type Body = {
+  provider: PaymentProviders;
+  checkoutId: string;
+  totalAmount: number;
+  captureAmount?: number; // support for partial payments
+};
+
+export type MollieResponse = {
+  provider: "mollie";
+  data: {
+    checkoutUrl: string;
+  };
+};
+
+export type SuccessResponse = {
+  provider: PaymentProviders;
+  ok: true;
+} & MollieResponse;
```

Comment:
Can we reuse global `PaymentProviderID` from `types/common.ts` instead of defining another similar type here for the backend, so we will have consistency with dashboard configuration, frontend, and backend?